tests/test_compatibility.py

```diff
@@ -0,0 +1,139 @@
+from tests.output_betterproto.oneof import Test as OneofTest
+from tests.output_betterproto.nested import Test as NestedTest, TestMsg as NestedTestMsg
+
+from betterproto import write_varint, read_varint
+
+from subprocess import run
+from shutil import which
+from pathlib import Path
+
+import pytest
+
+
+correct_oneof = OneofTest().from_dict(
+    {"pitied": 1, "just_a_regular_field": 123456789, "bar_name": "Testing"}
+)
+
+correct_nested = NestedTest().from_dict(
+    {
+        "nested": {"count": 1},
+        "sibling": {"foo": 2},
+        "sibling2": {"foo": 3},
+        "msg": NestedTestMsg.THIS,
+    }
+)
+
+
+@pytest.fixture(scope="module")
+@pytest.mark.skipif(which("mvn") is None, "Maven is absent and is required")
+@pytest.mark.skipif(which("jar") is None, "`jar` command is absent and is required")
+def compile_jar(output_dir):
+    # Compile the JAR
+    proc_maven = run(["mvn", "clean", "install", "-f", "tests/streams/java/pom.xml"])
+    if proc_maven.returncode != 0:
+        pytest.skip("Maven compatibility-test.jar build failed")
+
+
+@pytest.fixture(scope="module")
+def output_dir():
+    path = Path("tests/output_streams/")
+    if not path.exists():
+        path.mkdir(parents=True)
```

Comment:
This can be simplified to
```suggestion
    Path("tests/output_streams/").mkdir(parents=True, exist_ok=True)
```

Reply:
We also need to add `output_streams` to `.gitignore`

Reply:
Good point. I'd deliberately not used `exist_ok` as I remembered it not being present in earlier versions and that causing issues, but I think that was on one of the `os` functions so this seems to work perfectly.

Reply:
Fixed

Reply:
I think we should use `exist_ok=True` instead of `.exists()`